pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -189,6 +196,235 @@ func getV1Registries(config *tomlConfig) ([]Registry, error) {
 	return registries, nil
 }
 
+// registryIsMatch returns true if flags in the registry entry indicate that
+// its URL might not be used as-is, but will have URLs matched against it in
+// order to selectively apply configuration flags to an unlisted URL.
+func registryIsMatch(reg Registry) bool {
+	return reg.Insecure || reg.Blocked
+}
+
+// registryMatches returns true if the candidate location matches the pattern.
+// Recognized patterns include:
+// * IPv4[/class][:port][/path]
+// * IPv6[/class][:port][/path]
+// * hostname[:port][/path]
+// where IPv4, IPv6, and hostname may include wildcard characters.  The path
+// is implicitly terminated with "/", and individual components may also
+// include wildcard characters.
+func registryMatches(pattern, candidate string) (bool, error) {
+	defaultRegistryPort := "5000"
```

Comment:
The actual default is 443/80 - and Docker’s matching _does not_ treat the implied port and an explicit expression of the same port value as equivalent.

It can’t really treat them as equivalent because the 443/80 distinction only happens way later, after the “insecure” flag is 1) looked up 2) determined 3) an actual https connection attempt is made to try 443 even if marked as insecure.